File: src/main/java/forpleuvoir/hiirosakura/client/command/ServerChatMessageRegexCommand.java
package forpleuvoir.hiirosakura.client.command;

import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.arguments.StringArgumentType;
import com.mojang.brigadier.context.CommandContext;
import forpleuvoir.hiirosakura.client.config.HiiroSakuraDatas;
import net.fabricmc.fabric.api.client.command.v1.FabricClientCommandSource;

import static forpleuvoir.hiirosakura.client.command.base.HiiroSakuraClientCommand.*;
import static net.fabricmc.fabric.api.client.command.v1.ClientCommandManager.argument;
import static net.fabricmc.fabric.api.client.command.v1.ClientCommandManager.literal;

/**
 * 聊天消息正则匹配相关指令
 *
 * @author forpleuvoir
 * <p>#project_name hiirosakura
 * <p>#package forpleuvoir.hiirosakura.client.command
 * <p>#class_name ServerChatMessageRegexCommand
 * <p>#create_time 2021/6/24 0:33
 */
public class ServerChatMessageRegexCommand {
    public static final String TYPE = "scmr";

    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
        dispatcher.register(literal(COMMAND_PREFIX + TYPE)
                                    .then(literal("set")
                                                  .then(argument("regex", StringArgumentType.string())
                                                                .executes(ServerChatMessageRegexCommand::set)
                                                  )
                                    )
                                    .then(literal("remove")
                                                  .executes(ServerChatMessageRegexCommand::remove
                                                  )
                                    )
        );
    }

    private static int set(CommandContext<FabricClientCommandSource> context) {
        var regex = StringArgumentType.getString(context,"regex");
        HiiroSakuraDatas.SERVER_CHAT_MESSAGE_REGEX.put(regex);
        return 1;
    }

    private static int remove(CommandContext<FabricClientCommandSource> context) {
        HiiroSakuraDatas.SERVER_CHAT_MESSAGE_REGEX.remove();
        return 1;
    }
}

File: src/main/kotlin/forpleuvoir/hiirosakura/client/command/ServerChatMessageRegexCommand.kt
package forpleuvoir.hiirosakura.client.command

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.arguments.StringArgumentType
import com.mojang.brigadier.context.CommandContext
import forpleuvoir.hiirosakura.client.command.base.HiiroSakuraClientCommand.COMMAND_PREFIX
import forpleuvoir.hiirosakura.client.config.HiiroSakuraData
import net.fabricmc.fabric.api.client.command.v1.ClientCommandManager
import net.fabricmc.fabric.api.client.command.v1.FabricClientCommandSource

/**
 * 聊天消息正则匹配相关指令
 *
 * @author forpleuvoir
 *
 * #project_name hiirosakura
 *
 * #package forpleuvoir.hiirosakura.client.command
 *
 * #class_name ServerChatMessageRegexCommand
 *
 * #create_time 2021/6/24 0:33
 */
object ServerChatMessageRegexCommand {
	private const val TYPE = "scmr"
	fun register(dispatcher: CommandDispatcher<FabricClientCommandSource>) {
		dispatcher.register(ClientCommandManager.literal(COMMAND_PREFIX + TYPE)
			.then(ClientCommandManager.literal("set")
				.then(ClientCommandManager.argument("regex", StringArgumentType.string())
					.executes { set(it) }
				)
			)
			.then(ClientCommandManager.literal("remove")
				.executes { remove() }
			)
		)
	}

	private fun set(context: CommandContext<FabricClientCommandSource>): Int {
		val regex = StringArgumentType.getString(context, "regex")
		HiiroSakuraData.SERVER_CHAT_MESSAGE_REGEX.put(regex)
		return 1
	}

	private fun remove(): Int {
		HiiroSakuraData.SERVER_CHAT_MESSAGE_REGEX.remove()
		return 1
	}
}